fvm/state/accounts.go

```diff
@@ -29,6 +33,33 @@ type Accounts struct {
 	*addresses
 }
 
+// contractNames container for a list of contract names. Should always be sorted.
+// To ensure this, don't sort while reading it from storage, but sort it while adding/removing elements
+type contractNames []string
+
+func (l contractNames) Has(contractNames string) bool {
+	i := sort.SearchStrings(l, contractNames)
+	return i != len(l) && l[i] == contractNames
+}
+func (l *contractNames) add(contractNames string) {
+	i := sort.SearchStrings(*l, contractNames)
+	if i != len(*l) && (*l)[i] == contractNames {
+		// list already contains element
+		return
+	}
+	*l = append(*l, "")
+	copy((*l)[i+1:], (*l)[i:])
+	(*l)[i] = contractNames
+}
+func (l *contractNames) remove(contractName string) {
+	i := sort.SearchStrings(*l, contractName)
+	if i == len(*l) || (*l)[i] != contractName {
+		// list doesnt contain the element
+		return
+	}
+	*l = append((*l)[:i], (*l)[i+1:]...)
+}
```

Comment:
Can you move this code to the bottom of the file? Generally we try to keep the main constructor as the first entry, followed by receiver functions